tiledb/sm/stats/duration_instrument.h

```diff
@@ -27,79 +27,64 @@
  *
  * @section DESCRIPTION
  *
- * This file contains the TimerStat definition.
- *
- * This allows starting and stopping concurrent timers on different threads,
- * but does not count overlapping time periods toward the recorded duration.
- *
- * Thread-safety: none.
+ * This file defines class DurationInstrument.
  */
 
-#ifndef TILEDB_TIMER_STAT_H
-#define TILEDB_TIMER_STAT_H
+#ifndef TILEDB_DURATION_INSTRUMENT_H
+#define TILEDB_DURATION_INSTRUMENT_H
 
-#include <thread>
-#include <unordered_map>
+#include "tiledb/common/common.h"
+
+using namespace tiledb::common;
 
 namespace tiledb {
 namespace sm {
 namespace stats {
 
-class TimerStat {
+class Stats;
+
+/**
+ * This class contains a simple duration instrument.
+ */
+class DurationInstrument {
  public:
   /* ****************************** */
   /*   CONSTRUCTORS & DESTRUCTORS   */
   /* ****************************** */
 
-  /** Constructor. */
-  TimerStat();
+  /** Constructs an empty duration instrument object. */
```

Comment:
Is there a reason to keep this constructor? Right now, this constructor appears to lead to UB (at least) in the destructor because we try to call `parent_stats_->report_duration` unconditionally.